apollo-compiler/src/test/graphql/com/example/hero_details/adapter/HeroDetails_ResponseAdapter.kt

```diff
@@ -115,20 +123,24 @@ object HeroDetails_ResponseAdapter : ResponseAdapter<HeroDetails.Data> {
     }
 
     object FriendsConnection : ResponseAdapter<HeroDetails.Data.Hero.FriendsConnection> {
-      private val RESPONSE_FIELDS: Array<ResponseField> = arrayOf(
+      val RESPONSE_FIELDS: Array<ResponseField> = arrayOf(
         ResponseField(
           type = ResponseField.Type.Named.Other("Int"),
           responseName = "totalCount",
           fieldName = "totalCount",
           arguments = emptyMap(),
           conditions = emptyList(),
+          possibleFieldSets = emptyMap(),
         ),
         ResponseField(
           type = ResponseField.Type.List(ResponseField.Type.Named.Object("FriendsEdge")),
           responseName = "edges",
           fieldName = "edges",
           arguments = emptyMap(),
           conditions = emptyList(),
+          possibleFieldSets = mapOf(
+            "" to Edge.RESPONSE_FIELDS
+          ),
```

Comment:
have you considered to move this inside  `ResponseField.Type.Named.Object`? As only response field whose type is object can have selection set?

Reply:
Yup, will do 👍

Reply:
I didn't do that part yet as I remembered I put this comment: https://github.com/apollographql/apollo-android/blob/61a3372f6d25106e9011efaa700d3ec45ed6c36e/apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/ResponseField.kt#L56

The object vs non-object distinction feels a bit artificial. If we can remove `ResponseField.Type.Named.Object` ultimately, that'd be nice. Not sure we can but I'd like to explore that path a bit more before committing to the distinction